src/servers/ZoneServer2016/entities/npc.ts

```diff
@@ -89,15 +90,19 @@ export class Npc extends BaseFullCharacter {
       this.deathTime = Date.now();
       server.worldObjectManager.createLootbag(server, this);
       if (client) {
-        if (!server._soloMode) {
-          logClientActionToMongo(
-            server._db.collection(DB_COLLECTIONS.KILLS),
-            client,
-            server._worldId,
-            { type: "zombie" }
-          );
+        if (this.isZombie || this.isWildlife) {
+          const killType = this.isZombie ? "zombies" : "wildlife";
+
+          if (!server._soloMode) {
+            logClientActionToMongo(
+              server._db.collection(DB_COLLECTIONS.KILLS),
+              client,
+              server._worldId,
+              { type: killType }
+            );
+          }
+          (client.character.metrics as any)[killType + "Killed"]++;
```

Comment:
Nasty :P , well i would prefer an if or a switch it will almost add no processing and will have type safety

Reply:
Yea, when writing that I didn't like it but it was to reduce redundant code really, I changed it to something more type safe :P